models/daily_rates.sql

```diff
@@ -93,14 +119,16 @@ rates_w_overage as (
         ) as currency,
         base.usage_type like 'overage-%' as is_overage_rate,
         replace(base.usage_type, 'overage-', '') as associated_usage_type,
+        coalesce(remaining_balance_daily.is_account_in_overage, latest_remaining_balance_daily.is_account_in_overage) as _is_account_in_overage,
         case
-            when remaining_balance_daily.is_account_in_overage and is_overage_rate then 1
-            when not remaining_balance_daily.is_account_in_overage and not is_overage_rate then 1
+            when _is_account_in_overage and is_overage_rate then 1
+            when not _is_account_in_overage and not is_overage_rate then 1
             else 0
         end as rate_priority
```

Comment:
OOI what's the reason for doing this vs a seemingly simpler 'is_priority' field which is then just where filtered on below vs qualified

Reply:
the `is_priority` field would look something like:

```sql
        case
            when _is_account_in_overage and is_overage_rate then true
            when not _is_account_in_overage and not is_overage_rate then true
            else false
        end as is_priority
```

consider the scenario where an account is in overage but we haven't got an overage rate to choose from in `rate_sheet_daily` - the `is_priority` field would always be `false`, and we'd filter out rows for that day/service